api/src/main/java/org/apache/cloudstack/api/command/admin/storage/UpdateStoragePoolCmd.java

```diff
@@ -147,7 +147,22 @@ public void setUrl(String url) {
 
     @Override
     public void execute() {
-        StoragePool result = _storageService.updateStoragePool(this);
+        StoragePool result = null;
+        if (name != null ||
+                (tags != null && !tags.isEmpty()) ||
+                capacityIops != null ||
+                capacityBytes != null ||
+                (details != null && !details.isEmpty()) ||
+                url != null ||
+                isTagARule != null) {
```

Comment:
We could use ObjectUtils.anyNotNull() for all non collection variables and CollectionUtils.isNotEmpty() for details and tags.

Reply:
updated the code. Thanks @JoaoJandre